runtime/signals_nat.c

```diff
@@ -108,7 +108,8 @@ void caml_garbage_collection()
     alloc_bsize = whsize * sizeof(value);
 
     /* Put the young pointer back to what is was before our tiggering allocation */
-    Caml_state->young_ptr += alloc_bsize;
+    Caml_state->young_ptr = (value *)((uintnat)Caml_state->young_ptr
```

Comment:
I'm wondering if in this segment here, we do away with `alloc_bsize`
Then just do pointer arithmetic with `whsize` on the raw `value` pointers in `young_ptr` and `young_limit`. 
That should cut the amount of casting things around.

Reply:
Right, good point.
I pushed the change just now, thank you for the suggestion!